Sources/Core/AWSClientRuntime/AWSClientConfigDefaultsProvider.swift

```diff
@@ -19,10 +19,6 @@ public class AWSClientConfigDefaultsProvider {
     public static let idempotencyTokenGenerator: IdempotencyTokenGenerator
         = RuntimeConfigType.defaultIdempotencyTokenGenerator
 
-    public static func logger(_ clientName: String) -> LogAgent {
-        return RuntimeConfigType.defaultLogger(clientName: clientName)
-    }
```

Comment:
Q: Am I correct in saying that this `func logger` is refactored to `ExtractTelemetryLoggerConfig` in smithy-swift PR? I.e., from runtime default providing function to codegen time `SwiftIntegration` that writes default value in initializer.

Reply:
> from runtime default providing function to codegen time SwiftIntegration that writes default value in initializer.

Yes, the default value is written through the configured `TelemetryProvider`, e.g. `config.telemetryProvider.getLoggerProvider.getLogger`.

Previously, each client configuration would create a default logger via `AWSClientConfigDefaultsProvider.logger`, which was a re-export of `DefaultSDKRuntimeConfiguration.logger`.

In an earlier iteration, `config.telemetryProvider.getLoggerProvider.getLogger` every time a logger was used, but that would create a logger per operation call which is expensive, e.g. https://github.com/awslabs/aws-sdk-swift/blob/454ad2b1de7b5f27fe74b88d4030c55a230c8e62/Sources/Services/AWSAccessAnalyzer/AccessAnalyzerClient.swift#L164

So, instead I opted to keep an `internal` default logger on the configuration that could be used as the default `logger` in operations (and [presigning](https://github.com/awslabs/aws-sdk-swift/blob/454ad2b1de7b5f27fe74b88d4030c55a230c8e62/Sources/Services/AWSS3/models/Models.swift#L8859), otherwise I would have made it `fileprivate`). This would NOT be configurable directly via `config.logger`.

There are definitely alternatives to the approach, but I found this made sense.

Reply:
I see, that makes sense.